packages/zowe-explorer/src/Profiles.ts

```diff
@@ -1237,6 +1238,119 @@ export class Profiles extends ProfilesCache {
         }
     }
 
+    public async basicAuthClearSecureArray(profileName?: string): Promise<void> {
+        const profAttrs = await this.getProfileFromConfig(profileName);
+        const configApi = (await this.getProfileInfo()).getTeamConfig();
+        configApi.set(`${profAttrs.profLoc.jsonLoc}.secure`, ["tokenValue"]);
+        configApi.delete(`${profAttrs.profLoc.jsonLoc}.properties.user`);
+        configApi.delete(`${profAttrs.profLoc.jsonLoc}.properties.password`);
```

Comment:
Related to Fernando's comment, when switching from basic auth to token auth this won't clear user/password if present in the base profile.

Instead of hardcoding the JSON location where the secure properties are stored, we may want to use the `configApi.api.secure.securePropsForProfile` API documented [here](https://docs.zowe.org/stable/typedoc/classes/_zowe_imperative.configsecure.html#securepropsforprofile) and warn/prompt the user before removing properties from a base profile 😋